app/models/detailed_guide.rb

```diff
@@ -145,18 +138,54 @@ def published_inbound_related_detailed_guides
     DetailedGuide.published.joins(:outbound_edition_relations).where(edition_relations: { document_id: document.id })
   end
 
-  def related_mainstream_content_valid?
-    if related_mainstream_content_url.present? && related_mainstream_content_title.blank?
-      errors.add(:related_mainstream_content_title, "cannot be blank if a related URL is given")
+  def related_mainstream_found
+    return unless related_mainstream_requested?
+    fetch_related_mainstream_content_ids
+    add_errors_for_missing_related_mainstream
+    add_errors_for_missing_additional_related_mainstream
+  end
+
+  def add_errors_for_missing_related_mainstream
+    if @content_ids.count >= 1
+      errors.add(:related_mainstream_content_url, "This mainstream content could not be found") if @content_ids[0].nil?
     end
   end
 
-  def additional_related_mainstream_content_valid?
-    if additional_related_mainstream_content_url.present? && additional_related_mainstream_content_title.blank?
-      errors.add(:additional_related_mainstream_content_title, "cannot be blank if an additional related URL is given")
+  def add_errors_for_missing_additional_related_mainstream
+    if @content_ids.count > 1
+      errors.add(:additional_related_mainstream_content_url, "This mainstream content could not be found") if @content_ids[1].nil?
     end
   end
 
+  def fetch_related_mainstream_content_ids
+    base_paths = [related_mainstream_base_path, additional_related_mainstream_base_path].compact
+    if base_paths.any?
+      @content_ids ||= lookup_content_ids(base_paths)
+    else
+      @content_ids ||= []
+    end
+  end
+
+  def lookup_content_ids(base_paths)
+    @content_ids = []
+    base_paths.each do |base_path|
```

Comment:
It could be indeed! However when querying two base_paths, one correct and one incorrect, I want the response to look like `[nil, "a_content_id"]` so that I can infer whether the error should be displayed on the `related_mainstream_content_url` field or the `additional_related_mainstream_content_url`. That's what I do in the [add_errors_for_missing_related_mainstream](https://github.com/alphagov/whitehall/pull/2747/files#diff-cb1211485f1a19640c777ed86bf0b686R146) method.

If I used `Whitehall.publishing_api_v2_client.lookup_content_ids(base_paths).values`, the answer would look like `["a_content_id"]` and I couldn't tell which field the error belongs to.